core/templates/dev/head/domain/collection/CollectionUpdateService.js

```diff
@@ -120,39 +115,51 @@ oppia.factory('CollectionUpdateService', [
     };
 
     // A mutator object which provides generic apply() and reverse() functions
-    // for changing the skills of a collection. The skillListGetterCallback()
-    // takes a CollectionNode domain object and needs to return a SkillList
-    // domain object whose skills can be updated. The oldSkills parameter must
-    // be a copy of the current skills stored in the SkillList domain object.
-    var SetSkillsMutator = function(oldSkills, skillListGetterCallback) {
+    // for changing the skill IDs of a collection. The skillIdsGetterCallback()
+    // takes a CollectionNode domain object and needs to return a list of skill
+    // IDs whose elements can be updated. The oldSkillIds parameter must be a
+    // copy of the current skill IDs.
+    var SetSkillIdsMutator = function(oldSkillIds, skillIdsGetterCallback) {
       var _getCollectionNode = function(changeDict, collection) {
         var explorationId = _getExplorationIdFromChangeDict(changeDict);
         return collection.getCollectionNodeByExplorationId(explorationId);
       };
       this.apply = function(changeDict, collection) {
-        var skills = _getNewPropertyValueFromChangeDict(changeDict);
+        var newSkillIds = _getNewPropertyValueFromChangeDict(changeDict);
         var collectionNode = _getCollectionNode(changeDict, collection);
-        skillListGetterCallback(collectionNode).setSkills(skills);
+        var skillIds = skillIdsGetterCallback(collectionNode);
+        // Empty the list.
+        skillIds.length = 0;
+        // Add new skill IDs to the list without changing its binding.
```

Comment:
That was originally why I broke it out, yep. But I'll see if I can encapsulate this within a method in the collection node.